core/server/errors/no-database-found.js

```diff
@@ -0,0 +1,11 @@
+function NoDatabaseFound(message) {
+    this.message = message;
+    this.stack = new Error().stack;
+    this.statusCode = 404;
+    this.errorType = this.name;
+}
+
+NoDatabaseFound.prototype = Object.create(Error.prototype);
+NoDatabaseFound.prototype.name = 'NoDatabaseFound';
+
+module.exports = NoDatabaseFound;
```

Comment:
IMO this should be using InternalServerError? 404 is not the right status code as that indicates the client made a mistake in requesting, whereas anything to do with the database is a server-side error.